stand/efi/loader/arch/amd64/Makefile.inc

```diff
@@ -1,15 +1,40 @@
 
+.PATH: ${BOOTSRC}/i386/libi386
+SRCS+=	nullconsole.c \
+	comconsole.c \
+	spinconsole.c
+
+CFLAGS+=	-fPIC
+LDFLAGS+=	-Wl,-znocombreloc
+
+.if ${DO32:U0} == 0
 SRCS+=	amd64_tramp.S \
 	start.S \
 	elf64_freebsd.c \
 	trap.c \
 	multiboot2.c \
 	exc.S
 
-.PATH:	${BOOTSRC}/i386/libi386
-SRCS+=	nullconsole.c \
-	comconsole.c \
-	spinconsole.c
+SUBDIR.${MK_LOADER_IA32}+=	../loader_${LOADER_INTERP}-ia32
 
-CFLAGS+=	-fPIC
-LDFLAGS+=	-Wl,-znocombreloc
+.elif ${MK_LOADER_IA32} == "yes"
+.PATH: ${.CURDIR}/../loader/arch/amd64/ia32
+
+LOADER:=	${LOADER}-ia32
+NEWVERSWHAT=	"EFI loader" amd64-ia32
+CFLAGS+=	-I${EFISRC}/include/i386
+
+LDSCRIPT=	${.CURDIR}/../loader/arch/amd64/ia32/i386.ldscript
+.if ${LOADER_INTERP} == ${LOADER_DEFAULT_INTERP}
+LINKS=	${BINDIR}/${LOADER}.efi ${BINDIR}/loader-ia32.efi
+.endif
+
+SRCS+=	amd64_tramp32.S \
+	start32.S \
+	elf64_freebsd32.c
+
+LIBEFI32=	${BOOTOBJ}/efi/libefi32/libefi.a
+
+DPADD=	${LDR_INTERP32} ${LIBEFI32} ${LIBSA32} ${LDSCRIPT}
+LDADD=	${LDR_INTERP32} ${LIBEFI32} ${LIBSA32}
+.endif
```

Comment:
This seems alll wrong, though. It seems like we should use stnad/efi/loader/arch/i386/Makefile.inc when building the ia32 version.

Reply:
Yes that would make more sense now that you mention it.
When I first wrote this patch, the arch/i386 dir was being used for the half-completed 32-bit loader.